src/app/dashboard/views/DiscoveryAddForm/DiscoveryAddForm.test.tsx

```diff
@@ -84,61 +141,68 @@ describe("DiscoveryAddForm", () => {
     state.subnet.loaded = false;
     state.vlan.loaded = false;
     const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter
-          initialEntries={[{ pathname: "/dashboard", key: "testKey" }]}
-        >
-          <CompatRouter>
-            <DiscoveryAddForm discovery={discovery} onClose={jest.fn()} />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
+    renderWithBrowserRouter(
+      <DiscoveryAddForm discovery={discovery} onClose={jest.fn()} />,
+      { route: "/dashboard", wrapperProps: { store } }
     );
-    expect(wrapper.find("Spinner").exists()).toBe(true);
+    expect(screen.getByText("Loading")).toBeInTheDocument();
   });
 
-  it("maps name errors to hostname", () => {
```

Comment:
@huwshimi Could you take a look at this one for me? With RTL we can't directly access this prop like the original test tries to, and the only visual way of telling there's an error is to see if the save button is disabled

Reply:
This was a bit confusing to figure out. What this test is intended to do is test that name errors returned by the API are displayed (by mapping them to the hostname field). However, I discovered a little issue with how the formik errors are handled so that the errors in state weren't being displayed on first render. I think this is only an issue when we're writing a test like this.

We could potentially fix it by updating [useFormikErrors](https://github.com/canonical/maas-ui/blob/main/src/app/base/hooks/forms.ts#L19) to not store the initial value of the errors like this `const previousErrors = usePrevious(errors, false);`, but I'm not sure if that's intentionally storing the value for some reason.

The other solution would be to work around this by rerendering the form with the new state to simulate an update from the API. Something like this:

```typescript
it("maps name errors to hostname", async () => {
  // Render the form with default state.
  const { rerender } = renderWithBrowserRouter(
    <DiscoveryAddForm discovery={discovery} onClose={jest.fn()} />,
    { route: "/dashboard", wrapperProps: { state } }
  );
  const error = "Name is invalid";
  // Change the device state to included the errors (as if it has changed via an API response).
  state.device.errors = { name: error };
  // Rerender the form to simulate the state change.
  rerender(<DiscoveryAddForm discovery={discovery} onClose={jest.fn()} />);
  expect(
    screen.getByRole("textbox", {
      name: `${FormFieldLabels.Hostname} (optional)`,
    })
  // react-components uses aria-errormessage to link the errors to the inputs so we can use the toHaveErrorMessage helper here.
  ).toHaveErrorMessage(`Error: ${error}`);
});
```

Does that make sense? Let me know if not!

Reply:
That works perfectly! And yeah that makes a lot of sense, that's pretty clean